clickhouse_driver/log.py

```diff
@@ -30,9 +30,14 @@ def log_block(block):
         # thread_number in servers prior 20.x
         thread_id = row.get('thread_id') or row['thread_number']
 
+        host_name = row.get('host_name')
+        if host_name:
+            host_name = '[ {} ] '.format(host_name)
+
         logger.info(
-            '{%s} [ %s ] <%s> %s: %s',
+            '{%s} %s[ %s ] <%s> %s: %s',
```

Comment:
`host_name` is always present in the `row`. We can use it directly in `logger.info`:

```python
'{%s} [ %s ][ %s ] <%s> %s: %s',
...
row['host_name']
```

Reply:
Yeah, I also dislike this extra concatenation, but I saw the same in the clickhouse sources, I guess that it was not always presented in the block:

https://github.com/ClickHouse/ClickHouse/blob/13d22bcd422f65141a5ddea7a75d1bd332b054c5/src/DataStreams/InternalTextLogsRowOutputStream.cpp#L37

Added in a0e9d17bfdfa4a65f14de2a9c885a438c1c43b92 (v1.1.54386-testing-178-ga0e9d17bfd)

Reply:
(I though about adding a comment but by some reason prefer not to, sigh)

Reply:
Okay. It seems that `query_id` is [also optional](https://github.com/ClickHouse/ClickHouse/blob/13d22bcd422f65141a5ddea7a75d1bd332b054c5/src/DataStreams/InternalTextLogsRowOutputStream.cpp#L69).

We have two options to keep things consistent here:
* to make `query_id` optional;
* to make `host_name` not optional.

AFAIK `query_id` always should be present. I think empty strings handling is done for pretty formatting in ClickHouse server's sources.

Let's try to make `host_name` not optional. What do you think?

Reply:
> Let's try to make host_name not optional. 

Agree